Amplify/Categories/DataStore/Model/Internal/Persistable.swift

```diff
@@ -67,4 +67,175 @@ struct PersistableHelper {
             return false
         }
     }
+
+    public static func isEqual(_ lhs: Any?, _ rhs: Persistable?) -> Bool {
+        if lhs == nil && rhs == nil {
+            return true
+        }
+        switch (lhs, rhs) {
+        case let (lhs, rhs) as (Bool, Bool):
+            return lhs == rhs
+        case let (lhs, rhs) as (Temporal.Date, Temporal.Date):
+            return lhs == rhs
+        case let (lhs, rhs) as (Temporal.DateTime, Temporal.DateTime):
+            return lhs == rhs
+        case let (lhs, rhs) as (Temporal.Time, Temporal.Time):
+            return lhs == rhs
+        case let (lhs, rhs) as (Double, Double):
+            return lhs == rhs
+        case let (lhs, rhs) as (Int, Int):
+            return lhs == rhs
+        case let (lhs, rhs) as (Int, Double):
+            return Double(lhs) == rhs
+        case let (lhs, rhs) as (Double, Int):
+            return lhs == Double(rhs)
+        case let (lhs, rhs) as (String, String):
+            return lhs == rhs
+        default:
+            return false
+        }
+    }
+
+    public static func isLessOrEqual(_ lhs: Any?, _ rhs: Persistable?) -> Bool {
+        if lhs == nil && rhs == nil {
+            return true
+        }
+        switch (lhs, rhs) {
+        //case Bool Removed
+        case let (lhs, rhs) as (Temporal.Date, Temporal.Date):
+            return lhs <= rhs
+        case let (lhs, rhs) as (Temporal.DateTime, Temporal.DateTime):
+            return lhs <= rhs
+        case let (lhs, rhs) as (Temporal.Time, Temporal.Time):
+            return lhs <= rhs
+        case let (lhs, rhs) as (Double, Double):
+            return lhs <= rhs
+        case let (lhs, rhs) as (Int, Int):
+            return lhs <= rhs
+        case let (lhs, rhs) as (Int, Double):
+            return Double(lhs) <= rhs
+        case let (lhs, rhs) as (Double, Int):
+            return lhs <= Double(rhs)
+        case let (lhs, rhs) as (String, String):
+            return lhs <= rhs
+        default:
+            return false
+        }
+    }
+
+    public static func isLessThan(_ lhs: Any?, _ rhs: Persistable?) -> Bool {
```

Comment:
nit: maybe we can add a comment here and to the other methods about promoting `Int`s to `Double` in case of `Int`/`Double` or `Double`/`Int` comparison?